homeassistant/components/modbus/sensor.py

```diff
@@ -207,6 +228,10 @@ def update(self):
             return
         byte_string = b"".join([x.to_bytes(2, byteorder="big") for x in registers])
         val = struct.unpack(self._structure, byte_string)[0]
-        self._value = format(
-            self._scale * val + self._offset, ".{}f".format(self._precision)
-        )
+        val = self._scale * val + self._offset
+        if isinstance(val, int):
+            self._value = str(val)
+            if self._precision > 0:
+                self._value += "." + "0" * self._precision
+        else:
+            self._value = format(val, ".{}f".format(self._precision))
```

Comment:
Please use f-strings instead to format the string. We prefer that now over `str.format`.

Reply:
Ok